src/js/components/Button/Button.js

```diff
@@ -350,6 +350,12 @@ const Button = forwardRef(
         });
     }
 
+    if (buttonIcon && theme.icon?.matchSize && !buttonIcon.props.size) {
```

Comment:
Would it make any sense to combine this into a custom hook, such that we could do:
` const icon = useSizedIcon(iconProp, size);`
That might cut down on re-rendering a bit.